File: labs/lab01/AquecimentoRecursivo.java
package labs.lab01;

import java.util.Scanner;

public class AquecimentoRecursivo {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String string;
        while(!(string = scanner.nextLine()).equals("FIM")){
            int num = contar (string, 0);
            System.out.println(num);
            
        }
        scanner.close();
    }

    public static int contar(String string, int i) {
        if (i >= string.length()) {
            return 0;
        } else {
            if (string.charAt(i) >= 'A' && string.charAt(i) <= 'Z') {
                return 1 + contar(string, i + 1);
            } else {
                return contar(string, i + 1);
            }
        }
        
    }
}

File: labs/lab01/AquecimentoRecursivo.c
#include <stdio.h>
#include <string.h>

int contar(char string[], int i) {
    if (string[i] == '\0') {
        return 0;
    } else {
        if (string[i] >= 'A' && string[i] <= 'Z') {
            return 1 + contar(string, i + 1);
        } else {
            return contar(string, i + 1);
        }
    }
}

int main() {

    char palavra[1000];
    while (1) 
    {
        scanf(" %[^\n]", palavra);

        if (strcmp(palavra, "FIM") == 0) {
            break;
        } else {
            int num = contar(palavra, 0);
            printf("%d\n", num); 
        }
    }
    
    return 0; 
}
